spec/models/post_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  # Create post
+
+  subject { Post.create(author: User.new(name: 'morris'), title: 'test_title', text: 'test text') }
+
+  it 'should not allow empty title' do
+    expect(subject).to be_valid
+    subject.title = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'CommentsCounter should be bigger or equal than 0' do
+    expect(subject.comments_counter).to be >= 0
+    subject.comments_counter = -1
+  end
+
+  it 'LikesCounter should be bigger or equal than 0' do
+    expect(subject.likes_counter).to be >= 0
+  end
+
+  it 'UpdatePostsCounter to increase on new posts' do
+    expect(subject.author.posts_counter).to be 1
+  end
+
+  it 'FiveMostRecentComments length should be between 0 and 5' do
+    expect(subject.five_most_recent_comments).to be_kind_of Array
+    expect(subject.five_most_recent_comments.length).to be_between(0, 5)
+  end
```

Comment:
- [ ] You do not have a `five_most_recent_comments` in your Post model. What you have is `top_five`. Kindly rectify that.

- [ ] Also that method does not really return an Array. It more like returns an ActiveRelation, so the first expectation fails. You can change to something like this:

  ```
  it 'FiveMostRecentComments length should be between 0 and 5' do
    expect(subject.top_five.length).to be_between(0, 5)
  end
  ```